Lib/test/libregrtest/main.py

```diff
@@ -311,10 +312,24 @@ def rerun_failed_tests(self):
 
         self.log()
         self.log("Re-running failed tests in verbose mode")
-        self.rerun = self.bad[:]
-        for test_name in self.rerun:
-            self.log(f"Re-running {test_name} in verbose mode")
+        rerun_list = self.rerun[:]
+        self.rerun = []
+        for result in rerun_list:
+            test_name = result.name
+            errors = result.errors or []
+            failures = result.failures or []
+            error_names = [e[0].split(" ")[0] for e in errors]
+            failure_names = [f[0].split(" ")[0] for f in failures]
```

Comment:
I had some trouble following what `e[0].split(" ")[0]` is doing. What about:


```suggestion
            error_names = [test_name.split(" ")[0] for (test_name, *_) in errors]
            failure_names = [test_name.split(" ")[0] for (test_name, *_) in failures]
```

Reply:
`test_name` in often used in libregrtest to denote the name of the test file. I'll use `method_name` instead.